server/channels/web/handlers.go

```diff
@@ -378,38 +379,8 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// Handle errors that have occurred
 	if c.Err != nil {
-		c.Err.RequestId = c.AppContext.RequestId()
-		c.LogErrorByCode(c.Err)
-		// The locale translation needs to happen after we have logged it.
-		// We don't want the server logs to be translated as per user locale.
-		c.Err.Translate(c.AppContext.T)
-
-		c.Err.Where = r.URL.Path
-
-		// Block out detailed error when not in developer mode
-		if !*c.App.Config().ServiceSettings.EnableDeveloper {
-			c.Err.WipeDetailed()
-		}
-
-		// Sanitize all 5xx error messages in hardened mode
-		if *c.App.Config().ServiceSettings.ExperimentalEnableHardenedMode && c.Err.StatusCode >= 500 {
-			c.Err.Id = ""
-			c.Err.Message = "Internal Server Error"
-			c.Err.WipeDetailed()
-			c.Err.StatusCode = 500
-			c.Err.Where = ""
-		}
-
-		if IsAPICall(c.App, r) || IsWebhookCall(c.App, r) || IsOAuthAPICall(c.App, r) || r.Header.Get("X-Mobile-App") != "" {
-			w.WriteHeader(c.Err.StatusCode)
-			w.Write([]byte(c.Err.ToJSON()))
-		} else {
-			utils.RenderWebAppError(c.App.Config(), w, r, c.Err, c.App.AsymmetricSigningKey())
-		}
-
-		if c.App.Metrics() != nil {
-			c.App.Metrics().IncrementHTTPError()
-		}
+		h.handleContextError(c, w, r)
```

Comment:
Can't comment above, but we are also returning in L306. Should we also have `h.handleContextError` there? In short, if we need to add `h.handleContextError` before every return, then it might be good to declare it in a `defer` clause right after declaring the context so that future `return` statements don't forget to add it.

Reply:
No, its not needed there or everywhere. `handleContextError` isn't a new logic, I created the function from existing code block in an `if` statement, so it wasn't being called previously. The flow is the same as before.